### PR TITLE
"Error occurred while loading images" for no apparent reason #2413

### LINKED ISSUE #2410: "Error occurred while loading images" for no apparent reason
1. Explore 
2. Search for something
3. Scroll
4. This message appears from time to time

Latest alpha

### PR DESCRIPTION
**Pull request by prajurock**
**"Error occurred while loading images" for no apparent reason**

Fixes #2410

What changes did you make and why?
Checked in handler error.

**Tests performed (required)**

Tested on {Asus Zenfone Max M1 pro} with API level 27

### CODE REVIEW

**Comment by prajurock** [comment]
@nicolas-raoul Please review this.

**Comment by nicolas-raoul** [comment]
Sorry I don't have any time left for reviews until summer :-/

**Comment by prajurock** [comment]
Okay

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -206,6 +206,16 @@
             android:label="@string/provider_bookmarks_location"
             android:syncable="false" />
 
+        <provider
```
Comment (maskaravivek) [inline app/src/main/AndroidManifest.xml]: Why do you need to add another file provider?

File: app/src/main/java/fr/free/nrw/commons/GenericFileProvider.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package fr.free.nrw.commons;
+
+import android.support.v4.content.FileProvider;
+
+public class GenericFileProvider extends FileProvider {
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/GenericFileProvider.java]: Can you explain how adding this would fix the issue?

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 3]
```diff
@@ -178,13 +179,15 @@ void shareScreen(Bitmap bitmap) {
         try {
             File file = new File(this.getExternalCacheDir(), "screen.png");
             FileOutputStream fOut = new FileOutputStream(file);
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, fOut);
+            bitmap.compress(Bitmap.CompressFormat.  PNG, 100, fOut);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: remove space

File: app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java [hunk 4]
```diff
@@ -178,13 +179,15 @@ void shareScreen(Bitmap bitmap) {
...
             fOut.flush();
             fOut.close();
             file.setReadable(true, false);
+            Uri photoURI = FileProvider.getUriForFile(this, this.getApplicationContext().getPackageName() + ".my.package.name.provider", file);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java]: Why is this needed? Our app already has a file provider defined.

**Comment by maskaravivek** [comment]
This has already been fixed
